addons/api/mirage/factories/credential.js

```diff
@@ -3,9 +3,20 @@ import { faker } from '@faker-js/faker';
 import permissions from '../helpers/permissions';
 import generatedId from '../helpers/id';
 
+const types = ['username_password', 'ssh_private_key'];
+
 export default factory.extend({
-  id: () => generatedId('cred_'),
-  type: 'username_password',
+  type: (i) => types[i % types.length],
+  id() {
+    switch (this.type) {
+      case 'ssh_private_key':
+        return generatedId('credspk_');
+      case 'username_password':
+        return generatedId('credup_');
+      default:
+        return generatedId('cred_');
```

Comment:
Are there types that return this currently? What else is there besides the `username_password` and `ssh_private_key`?

Reply:
Versions <0.10 will only have `cred_` so I included it as the default option. However, mirage doesn't care about the version so I can remove it. I typically add a default case for lint purposes but I don't think our lint rules care about that.

Reply:
Lint has no issue with a lack of default so I just removed it.